src/libstd/sync/rwlock.rs

```diff
@@ -171,14 +171,16 @@ impl<T: ?Sized> RwLock<T> {
         RwLockReadGuard::new(&*self.inner, &self.data)
     }
 
-    /// Attempts to acquire this lock with shared read access.
+    /// Attempts to acquire this rwlock with shared read access.
+    ///
+    /// If the access could not be granted at this time, then `Err` is returned.
+    /// Otherwise, an RAII guard is returned which will release the shared access
+    /// when it is dropped.
```

Comment:
I believe that a poisoned rwlock will also return an `Err` even if it was successful in the `try_read`, so technically the lock could be locked if `Err` is returned. Perhaps the docs could be updated slightly to account for this?